File: services/identity-service/src/main/java/br/com/deltaglobalbank/identity/infrastructure/security/ipAllowlist/TenantIpAllowlistProperties.java
package br.com.deltaglobalbank.identity.infrastructure.security.ipAllowlist;

import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.boot.context.properties.bind.DefaultValue;

@ConfigurationProperties(prefix = "identity.tenant-ip-allowlist")
public record TenantIpAllowlistProperties(
    @DefaultValue("300") long cacheTtlSeconds
) {
    public TenantIpAllowlistProperties {
        if (cacheTtlSeconds < 1 || cacheTtlSeconds > 3600) {
            throw new IllegalArgumentException(
                "identity.tenant-ip-allowlist.cache-ttl-seconds deve estar entre 1 e 3600");
        }
    }
}

File: services/identity-service/src/main/kotlin/br/com/deltaglobalbank/identity/infrastructure/security/ipAllowlist/TenantIpAllowlistProperties.kt
package br.com.deltaglobalbank.identity.infrastructure.security.ipAllowlist

import org.springframework.boot.context.properties.ConfigurationProperties

@ConfigurationProperties(prefix = "identity.tenant-ip-allowlist")
data class TenantIpAllowlistProperties(
    val cacheTtlSeconds: Long = 300
) {
    init{
        require(cacheTtlSeconds in 1..3600){
            "identity.tenant-ip-allowlist.cache-ttl-seconds deve estar entre 1 e 3600"
        }
    }
}